apps/checkout-app/backend/auth.ts

```diff
@@ -22,15 +51,15 @@ export const isAuthenticated = async (req: NextApiRequest) => {
     return false;
   }
 
-  const key = await jwksClient.getSigningKey();
-  const publicKey = key.getPublicKey();
+  const tokenData = decode(token);
 
-  try {
-    verify(token, publicKey);
-    return true;
-  } catch (err) {
+  if (typeof tokenData === "string" || !tokenData || !tokenData["iss"]) {
     return false;
   }
+
+  const jwtVerifier = JwtVerifier.getInstance(tokenData["iss"]);
```

Comment:
`tokenData["iss"]` is only used to verify the signature, but not as a domain for fetching private metadata. It's a security risk, as the attacker might use his own domain to pass the validation. **tokenData["iss"]** should be a domain constant used throughout the app, the app should not rely on hardcoded domain values